### DISCUSSION TITLE
Support for mss550l ?

### DISCUSSION THREAD

**Discussion by sysserwc** [krahabb/meross_lan#135] [Q&A]
Thanks for this excellent solution. I have a couple of plugs that are now managed through HA instead of Meross cloud. 

I also have a mss550l 2-way light switch that is not officially supported. Any chance of supporting this model in the future?

**Comment by krahabb** [comment]
Hello @sysserwc ,
I think it should be easy to support this switch too. You should be able to at least add it to HA the same way you added the others. If this happens we're halfway done.
Once the device is configured in meross_lan/HA you could already be able to use it (beside the brand name, I would guess Meross had implemented it like any other switch and so meross_lan should be able to use it)
If not, I would need a trace from the device in order to investigate it's own protocol nuances and try to implement a correct behaviour.
In order to trace the device you can follow https://github.com/krahabb/meross_lan#troubleshooting

**Reply by sysserwc** [reply]
Thanks for the response @krahabb ,

I tried connecting the switch and it seemed to work initially, but then started changing states randomly and switching the light on in the middle of the night, so I reverted back. 
Maybe because it is a 2-way it has another state for monitoring changes from the second switch? 

Anyway, I'll connect it back next week and upload the logs.
Thanks again!

**Reply by sysserwc** [reply]
Hi @krahabb ,
The mss550l is connected and so far seems to be working fine. Not sure why I had issues previously.
I will let it run for a few days and update if there are any issues. 
Thanks again for the quick response.
Cheers.

**Reply by krahabb** [reply]
Good to know!
At any rate if you can 'throw in' a disgnostic trace that would be helpful in order to check for possible issues (like the one you experienced before)
Thank you!

**Reply by sysserwc** [reply]
With pleasure. See trace below. Thank you for this great solution!

2021/12/27 - 11:55:19		auto	GETACK	Appliance.System.All	{"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606114, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640565000}], "triggerx": [], "timerx": []}}
2021/12/27 - 11:55:19		auto	GETACK	Appliance.System.Ability	{"Appliance.Config.Key": {}, "Appliance.Config.WifiList": {}, "Appliance.Config.Wifi": {}, "Appliance.Config.Trace": {}, "Appliance.Config.Install": {}, "Appliance.System.All": {}, "Appliance.System.Hardware": {}, "Appliance.System.Firmware": {}, "Appliance.System.Debug": {}, "Appliance.System.Online": {}, "Appliance.System.Time": {}, "Appliance.System.Clock": {}, "Appliance.System.Ability": {}, "Appliance.System.Runtime": {}, "Appliance.System.Report": {}, "Appliance.System.Position": {}, "Appliance.System.DNDMode": {}, "Appliance.Control.Multiple": {"maxCmdNum": 5}, "Appliance.Control.ToggleX": {}, "Appliance.Control.TimerX": {"sunOffsetSupport": 1}, "Appliance.Control.TriggerX": {}, "Appliance.Control.Bind": {}, "Appliance.Control.Unbind": {}, "Appliance.Control.Upgrade": {}, "Appliance.Digest.TriggerX": {}, "Appliance.Digest.TimerX": {}}
2021/12/27 - 11:55:19	TX	http	GET	Appliance.Config.Install	{"install": {}}
2021/12/27 - 11:55:19	RX	http	GETACK	Appliance.Config.Install	{"install": {"code": 0}}
2021/12/27 - 11:55:24	TX	http	GET	Appliance.System.Runtime	{"runtime": {}}
2021/12/27 - 11:55:24	RX	http	GETACK	Appliance.System.Runtime	{"runtime": {"signal": 89}}
2021/12/27 - 11:55:29	TX	http	GET	Appliance.System.Position	{"position": {}}
2021/12/27 - 11:55:29	RX	http	GETACK	Appliance.System.Position	{"position": {"longitude": 0, "latitude": 0}}
2021/12/27 - 11:55:30	TX	http	SET	Appliance.Control.ToggleX	{"togglex": {"channel": 0, "onoff": 1}}
2021/12/27 - 11:55:30	RX	http	SETACK	Appliance.Control.ToggleX	{}
2021/12/27 - 11:55:33	TX	http	SET	Appliance.Control.ToggleX	{"togglex": {"channel": 0, "onoff": 0}}
2021/12/27 - 11:55:33	RX	http	SETACK	Appliance.Control.ToggleX	{}
2021/12/27 - 11:55:34	TX	http	GET	Appliance.Control.ToggleX	{"togglex": {}}
2021/12/27 - 11:55:34	RX	http	GETACK	Appliance.Control.ToggleX	{"channel": 0, "togglex": {"channel": 0, "onoff": 0, "lmTime": 1640606132}}
2021/12/27 - 11:55:39	TX	http	GET	Appliance.Control.TimerX	{"timerx": {}}
2021/12/27 - 11:55:40	RX	http	GETACK	Appliance.Control.TimerX	{"digest": {}, "timerx": {}}
2021/12/27 - 11:55:44	TX	http	GET	Appliance.Control.TriggerX	{"triggerx": {}}
2021/12/27 - 11:55:44	RX	http	GETACK	Appliance.Control.TriggerX	{"digest": {}, "triggerx": {}}
2021/12/27 - 11:55:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 11:55:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606144, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 1, "lmTime": 1640606143}], "triggerx": [], "timerx": []}}}
2021/12/27 - 11:55:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 11:55:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 11:55:49	TX	http	GET	Appliance.Digest.TriggerX	{"digest": []}
2021/12/27 - 11:55:49	RX	http	GETACK	Appliance.Digest.TriggerX	{"digest": []}
2021/12/27 - 11:55:54	TX	http	GET	Appliance.Digest.TimerX	{"digest": []}
2021/12/27 - 11:55:54	RX	http	GETACK	Appliance.Digest.TimerX	{"digest": []}
2021/12/27 - 11:55:55	TX	http	SET	Appliance.Control.ToggleX	{"togglex": {"channel": 0, "onoff": 0}}
2021/12/27 - 11:55:55	RX	http	SETACK	Appliance.Control.ToggleX	{}
2021/12/27 - 11:56:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 11:56:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606174, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 11:56:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 11:56:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 11:56:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 11:56:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606204, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 11:56:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 11:56:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 11:57:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 11:57:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606234, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 11:57:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 11:57:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 11:57:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 11:57:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606264, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 11:57:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 11:57:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 11:58:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 11:58:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606294, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 11:58:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 11:58:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 11:58:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 11:58:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606324, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 11:58:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 11:58:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 11:59:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 11:59:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606354, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 11:59:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 11:59:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 11:59:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 11:59:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606384, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 11:59:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 11:59:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:00:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:00:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606414, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:00:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:00:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:00:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:00:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606444, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:00:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:00:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:01:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:01:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606474, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:01:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:01:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:01:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:01:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606504, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:01:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:01:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:02:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:02:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606534, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:02:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:02:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:02:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:02:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606564, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:02:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:02:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:03:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:03:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606594, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:03:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:03:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:03:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:03:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606624, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:03:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:03:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:04:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:04:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606654, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:04:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:04:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:04:45	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:04:45	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606684, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:04:45	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:04:45	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}
2021/12/27 - 12:05:15	TX	http	GET	Appliance.System.All	{"all": {}}
2021/12/27 - 12:05:15	RX	http	GETACK	Appliance.System.All	{"all": {"system": {"hardware": {"type": "mss550l", "subType": "us", "version": "2.0.0", "chipType": "mt7682", "uuid": "################################", "macAddress": "#################"}, "firmware": {"version": "2.1.6", "compileTime": "2019/06/13 14:05:24 GMT +08:00", "wifiMac": "#################", "innerIp": "#############", "server": "#############", "port": "####", "userId": "#"}, "time": {"timestamp": 1640606714, "timezone": "", "timeRule": []}, "online": {"status": 0}}, "digest": {"togglex": [{"channel": 0, "onoff": 0, "lmTime": 1640606165}], "triggerx": [], "timerx": []}}}
2021/12/27 - 12:05:15	TX	http	GET	Appliance.System.DNDMode	{"DNDMode": {}}
2021/12/27 - 12:05:15	RX	http	GETACK	Appliance.System.DNDMode	{"DNDMode": {"mode": 1}}

**Reply by krahabb** [reply]
Hello @sysserwc ,
It really looks, by inspecting the device 'abilities', this is really the same as any other switch device so it basically supports toggling on/off and the 'presence light/DNDMode' like any other smart plug or the likes.
I'll add it to the list of the 'offically' supported hardware one of these days.
Thank you for your contribution

**Comment by donparlor** [comment]
Hello @krahabb ! I have a 3-way Meross Switch [MSS550](https://www.amazon.ca/gp/product/B07Y9S9NQ7/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1). It must be quite similar to the MSS550l. 

After installing Meross-Lan from HACS, I can't find my switch in the entity list. Am I missing something? Did I forget a step?

**Comment by krahabb** [comment]
Hello @donparlor ,
The device should automatically be discovered when HA sees some dhcp related traffic on the network. If this is not happening (for whatever reason - maybe rebooting the device would fix this) you should manually add it by adding meross_lan from the Integrations page in HA. 
When doing so you could be offered a 'Meross LAN MQTT Hub' (this happens by default when you also have some MQTT integration configured in HA). Confirm this entry and proceed by adding another meross_lan integration with the same process.
This time you will be asked for the ip address and the device key

**Comment by donparlor** [comment]
Hello again @krahabb,

I properly installed the HACS for meross_lan and restarted my HA server.
<img width="502" alt="Capture d’écran, le 2022-01-08 à 12 27 47" src="https://user-images.githubusercontent.com/97048708/148653735-185ef2ce-b37a-4e7a-bf9b-4f23b933eda2.png">

When I go to the integration tab and look for any Meross integration, it doesn't show up.
<img width="540" alt="Capture d’écran, le 2022-01-08 à 12 28 12" src="https://user-images.githubusercontent.com/97048708/148653758-313c1664-82de-4b62-bfc5-4fdd2db25056.png">

Could there be a new issue arising with your integration?

**Reply by donparlor** [reply]
I autocorrect myself. It seems installing in HACS finally worked. I really don't know what made it work, but I have spent 5-6 hours testing different options and it finally showed up in the notification bar.

**Reply by krahabb** [reply]
You were probably fooled by the HA frontend UI not reloading its cache when something (like the list of integrations) changes on the backend. It happens to me too I'm unable to find what 'should be' there and the solution is to reload your browser UI by refreshing its page or so (depending on the browser - Ctrl-R on Chrome)

**Reply by donparlor** [reply]
What confused me came from a youtuber explaining how to install meross_lan. He speaks spanish but you can follow easily by looking at the video (https://youtu.be/kDWevQpnwnQ?t=317). At 5:17 he goes into HACS and install meross_lan, but then at 6:13 he goes into integration and search for meross_lan and there is a result which pops up, but I can't reproduce that step... On my side, for some mysterious reason, I finally had a notification after multiple restart, uninstall, re-install, restart, uninstall, re-install, restart, and by the end I was really about to give up when it popped up!

Thank you @krahabb, that's an amazing integration, it's just sad I couldn't find out what was happening earlier.
